macro/src/lib.rs

```diff
@@ -0,0 +1,403 @@
+// This file was originally based on cxx-async/macro/src/lib.rs from the `cxx-async` crate, which
+// is subject to the following copyright:
+//
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+// https://github.com/pcwalton/cxx-async
+//
+// The `cxx-async` crate is dual-licensed under both the MIT and Apache 2.0 licenses. You will find
+// a copy of one of those licenses in the file named LICENSE in this repository's root directory.
+//
+// Subsequent changes are subject to the following copyright:
+//
+// Copyright (c) 2025 Cloudflare, Inc.
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::parse::Parse;
+use syn::parse::ParseStream;
+use syn::parse::Result as ParseResult;
+use syn::spanned::Spanned;
+use syn::Error as SynError;
+use syn::Ident;
+use syn::ImplItem;
+use syn::ItemImpl;
+use syn::Lit;
+use syn::LitStr;
+use syn::Path;
+use syn::Result as SynResult;
+use syn::Token;
+use syn::Type;
+use syn::TypePath;
+use syn::Visibility;
+
+/// Defines a future type that can be awaited from both Rust and C++.
+///
+/// Invoke this macro like so:
+///
+/// ```ignore
+/// #[kj_rs::bridge_future]
+/// unsafe impl Future for RustFutureString {
+///     type Output = Result<String, Infallible>;
+/// }
+/// ```
+///
+/// Here, `RustFutureString` is the name of the future type declared inside the
+/// `extern Rust { ... }` block inside the `#[cxx::bridge]` module. The associated `Output` type
```

Comment:
It wasn't obvious to me that the macro would instantiate the type as a BoxFuture and the extern Rust block is (re-)referencing the type generated by this macro , rather than this macro referencing that generated type. Of course seeing how this went from the boilerplate to this macro makes it obvious, but I read this description first...

Reply:
Yeah, this was confusing to me, too. I think I can clarify.

Reply:
Reworded in https://github.com/capnproto/kj-rs/compare/86a329708be98ae0c4adee57e3bcd98f41a2c236..6782351be5844916f8f792a4f591ddcc0c8299aa

This comment was also incorrect, it seems -- the generated type needs to be used in an extern C++ block, rather than an extern Rust one.